### DISCUSSION TITLE
Difficulty Connecting to Remote

### DISCUSSION THREAD

**Discussion by EmperorKabir** [pybricks/support#1376] [Q&A] [answered]
Hi, I'm very limited in my coding knowledge and tried to get a remote control connected using the block coding tool. My intention is to run a custom program in an infinite loop when a button on the remote is pressed but if I instead press other buttons on the remote, the program will stop and I can manually control the motor. (I've essentially stuck a city hub and train motor in a lego hogwarts express). Obviously should the user choose to press one of the 'manual' buttons first, they will still operate the motor and if I press the magic program button, the loop program will begin again. 

That's the background but the main issue is that the remote simply won't connect. I tried restoring the Lego firmware and it connects fine and works fine, but in Pybricks, it just doesn't want to connect to the remote. I have tried disconnecting from Pybricks as well and restarting the hub but it still won't connect.

Quite possibly, I've just written awful code/bad block coding.
![Capture](https://github.com/pybricks/support/assets/114411808/a595f7fa-60a6-4741-b721-1d95924901e5)

**Comment by laurensvalk** [comment]
Can you try turning on the remote first, and then start your program (while the remote is still blinking)?

**Reply by EmperorKabir** [reply]
That didn't seem to do anything sadly. Video here for reference:

https://youtu.be/feuZrzz5B4U?si=NqTdMmL1i4L2lQLD

**Reply by laurensvalk** [reply]
Also, there's a few things in the program too. 

- Your program ends right away, so even if it did connect successfully, you wouldn't see it.
- Make sure to use the `plus` / `minus` buttons, which corresponds to the `+` and  `-` buttons on the remote.

Try something with a loop that runs forever, for example:

![image](https://github.com/pybricks/support/assets/12326241/a69f812d-546b-4142-bb1d-1257b90dd07c)

Let me know how this goes!

**Reply by EmperorKabir** [reply]
I tried a very similar set up but still no luck:

![image](https://github.com/pybricks/support/assets/114411808/0c3493b2-5e6e-4d2d-891f-e8f8d7510c41)

**Reply by laurensvalk** [reply, marked as answer]
Thanks for getting back. Note that there is no repeat block in your program.

Your program ends as soon as you start it, so nothing appears to happen. And so the remote disconnects right away, as it always does when a program ends.

Try making the suggested program above as a starting point, and build from there. :slightly_smiling_face:

**Reply by EmperorKabir** [reply]
Thank you! Seemed to help! I realie I have to be really sharp at pressing the 2 green buttons at the same time; there doesn't seem to be much wiggle room.

 I just have one other side question, (if this is the wrong place for it, I'm happy to post elsewhere if you could please signpost me but) the following is working but the only thing I want to amend is so that if while the more custom loop is running, I press left minus or left plus, the loop stops and those initial inputs work again. I can see there is a break loop option but I'm unsure how to adjust the block coding to get it to interact with the repeated loop. (That and maybe add a kill switch).
 
 
![image](https://github.com/pybricks/support/assets/114411808/137d355e-576f-4cd6-8a3f-14b653f39468)

**Reply by EmperorKabir** [reply]
Never mind, I figured it out with this! Thank for so much for your help! :)

I take it there is no way to make the train stop by pressing the same button again. (I changed it from remote right twice to the second one being remote minus). My next step will be figuring out whether I can increase timing tolerance between green button sync presses but I'm guessing not and that it's part of the firmware.

![image](https://github.com/pybricks/support/assets/114411808/09d7b50f-b66e-435e-a0a7-716d20572ae6)

**Reply by laurensvalk** [reply]
Glad to hear it's working! 

In principle you don't have to press both green buttons. You can just turn on the remote (blinking white) and then press the white/blue run button in the app to start your project.

You may be able to do what you want as suggested in the snippet above. You can expand the if/else blocks with extra else options using the white arrow at the bottom. Then you can basically make a case for each button as you like.

Look at it this way -- you just made your first real program :smile: Now you can try breaking it down in smaller pieces to make sure each block does what you want, and build from there.

**Reply by laurensvalk** [reply]
> I take it there is no way to make the train stop by pressing the same button again.

There is! It's just a different way of using the blocks. The possibilities are basically endless from here :)

Instead of doing if/else repeatedly, you could do repeatedly:

- wait until one or more buttons is pressed
- decide what to do based on which one (using if/else), for example turn the motor on
- wait for that button to be released again
- stop the motor

> My next step will be figuring out whether I can increase timing tolerance between green button sync presses but I'm guessing not and that it's part of the firmware.

Our messages might have just crossed. In principle no strict timing is needed, you can just start the program with the app.
